### PR TITLE
Problem: block number is not filled in eth logs

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- set block hash in tx response, so it can be read by json-rpc
- don't set block-hash/tx-hash/block-number in log, json-rpc can fill them instead
- remove the tx log event completely

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/statedb/statedb.go [hunk 1]
```diff
@@ -118,7 +118,6 @@ func (s *StateDB) AddLog(log *ethtypes.Log) {
 	s.journal.append(addLogChange{})
 
 	log.TxHash = s.txConfig.TxHash
-	log.BlockHash = s.txConfig.BlockHash
```
Comment (mmsqe) [inline x/evm/statedb/statedb.go:120]: I didn't know we can fill in api, we can save db size but will it be slower right?

Comment (yihuang) [reply x/evm/statedb/statedb.go:120]: most of the case, we don't add extra rpc queries, except get logs by block range

Comment (mmsqe) [reply x/evm/statedb/statedb.go:120]: seems eth filters also need

Comment (yihuang) [reply x/evm/statedb/statedb.go:120]: yeah, still looking into that, the websocket part

Comment (yihuang) [reply x/evm/statedb/statedb.go:120]: the other thing is, we seems duplicated the eth logs in tx response and events, both are contained in the tx result.
we can decode tx logs in the `Data` field directly, no need to search in event list.

Comment (mmsqe) [reply x/evm/statedb/statedb.go:120]: not sure if we could just fill in [FilterLogs](https://github.com/crypto-org-chain/ethermint/blob/31a2b472b714308f916418d87d52a73c1cb3a3aa/rpc/namespaces/ethereum/eth/filters/utils.go#L31) for 1st part

**Review by mmsqe — APPROVED** [review]
gladly approve

**Pull request merged**